x/poolmanager/types/keys.go

```diff
@@ -43,7 +44,16 @@ var (
 
 // ModuleRouteToBytes serializes moduleRoute to bytes.
 func FormatModuleRouteKey(poolId uint64) []byte {
-	return []byte(fmt.Sprintf("%s%d", SwapModuleRouterPrefix, poolId))
+	// Estimate the length of the string representation of poolId
+	// 7 is a safe upper bound, (9999999) pools, and is an 8 byte allocation (not bad)
```

Comment:
just for extra safety... should we check that poolId < 9999999?

Reply:
All we could really do is panic, which is what will already happen here anyway

Reply:
We can add that check to CreatePool

Reply:
Ok just raised the upperbound to `99,999,999,999`. At $1 to create a pool, this is a third the market cap of ETH, so going to go ahead and merge!